### PR TITLE
W-14185880: Make objects used by DataWeave part of the api

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/api/event/DefaultFlowCallStack.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.api.event;
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/api/event/DefaultFlowCallStack.java:7]: can't we expose an interface instead? or at most, make this priviledged

File: core/src/main/java/org/mule/runtime/core/api/el/dataweave/DataWeaveArtifactContext.java [hunk 2]
```diff
@@ -25,14 +24,33 @@
  *
  * @since 4.0
  */
-public class DataWeaveArtifactContext extends AbstractArtifactContext {
+public class DataWeaveArtifactContext {
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/api/el/dataweave/DataWeaveArtifactContext.java:27]: can't we expose an interface instead? or at most, make it priviledged

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/mule/runtime/core/api/event/DefaultFlowCallStack.java (hunk 1 above)
Comment (elrodro83) [reply core/src/main/java/org/mule/runtime/core/api/event/DefaultFlowCallStack.java:7]: DW doesn't handle interfaces, only concrete classes

File: core/src/main/java/org/mule/runtime/core/api/el/dataweave/DataWeaveArtifactContext.java (hunk 2 above)
Comment (elrodro83) [reply core/src/main/java/org/mule/runtime/core/api/el/dataweave/DataWeaveArtifactContext.java:27]: DW doesn't handle interfaces, only concrete classes

File: core/src/main/java/org/mule/runtime/core/privileged/event/DefaultFlowCallStack.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  }
+
+  @Override
+  public String toString() {
+    return doToString(FlowStackElement::toString);
+  }
+
+  public String toStringWithElapsedTime() {
+    return doToString(FlowStackElement::toStringWithElapsedTime);
+  }
```
Comment (mbuchwald) [inline core/src/main/java/org/mule/runtime/core/privileged/event/DefaultFlowCallStack.java:132]: Doc in this one?

**Review by mbuchwald — CHANGES_REQUESTED** [review]

**Review by mbuchwald — APPROVED** [review]

**Pull request merged**